processing/src/main/java/org/apache/druid/segment/Cursors.java

```diff
@@ -64,6 +64,26 @@ public static Order getTimeOrdering(final List<OrderBy> ordering)
     }
   }
 
+  public static boolean compatibleOrdering(CursorBuildSpec buildSpec, List<OrderBy> ordering)
+  {
+    // if the build spec doesn't prefer an ordering, any order is ok
+    if (buildSpec.getPreferredOrdering().isEmpty()) {
+      return true;
+    }
+    // all columns must be present in ordering if the build spec specifies them
+    if (ordering.size() < buildSpec.getPreferredOrdering().size()) {
+      return false;
+    }
+    for (int i = 0; i < buildSpec.getPreferredOrdering().size(); i++) {
+      final OrderBy preferredOrderBy = buildSpec.getPreferredOrdering().get(i);
```

Comment:
i don't imagine that it will ever be lazy since its just populated by a builder from the query engines, but this problem went away moving the function to `CursorBuildSpec`